src/components/interaction-designer/resource-editors/ModeResourceEditor.vue

```diff
@@ -0,0 +1,234 @@
+<template>
+  <!--Resource editors grouped by mode (channel)-->
+  <div v-if="resource"
+       class="mode-resource-editor d-flex flex-wrap">
+    <div v-for="({id: languageId, label: language}, languageIndex) in activeFlow.languages"
+         :key="languageId"
+         class="col-3">
+      <header class="d-flex">
+        <div class="mr-auto">{{language || trans('flow-builder.unknown-language')}}</div>
+      </header>
+
+      <template v-for="contentType in discoverContentTypesFor(mode)">
+        <resource-variant-text-editor
+          v-if="contentType === SupportedContentType.TEXT"
+          :index="computeResourceIndex(languageIndex, modeIndex)"
+          :mode="mode"
+          :resource-id="resource.uuid"
+          :resource-variant="findOrGenerateStubbedVariantOn(
+                  resource,
+                  {language_id: languageId, content_type: contentType, modes: [mode]})" />
+
+        <div v-if="contentType === SupportedContentType.AUDIO">
+          <validation-message
+            #input-control="{ isValid }"
+            :message-key="`resource/${resource.uuid}/values/${computeResourceIndex(languageIndex, modeIndex)}/value`">
+            <audio-library-selector
+              :audio-files="availableAudioFiles"
+              :lang-id="languageId"
+              :resource-id="resource.uuid"
+              :selected-audio-uri="findOrGenerateStubbedVariantOn(
+                      resource,
+                      {language_id: languageId, content_type: contentType, modes: [mode]}).value" />
+          </validation-message>
+
+          <phone-recorder
+            v-if="can(['edit-content', 'send-call-to-records'], true) && !findOrGenerateStubbedVariantOn(resource,{language_id: languageId, content_type: contentType, modes: [mode]}).value"
+            :recording-key="`${block.uuid}:${languageId}`" />
```

Comment:
I'm curious if `phone-recorder` should be included? (I don't know anything about it, just checking)

Reply:
Nevermind, I see it being present in the LanguageResourceEditor too :)